samples/AzCopyCore/AzCopyCore/Program.cs

```diff
@@ -212,7 +212,7 @@ static void TransferStorageFileToDirectory(ReadOnlySpan<char> storageFile, ReadO
                     }
                     index += length;
                     bytesLeft -= length;
-                    if (bytesLeft == 0) break;
+                    Debug.Assert(bytesLeft >= 0);
```

Comment:
I don't think this Debug.Assert is necessary, given the loop condition is already stronger. Maybe a better option would be to add a Debug.Assert after the loop exits, if we know that bytesLeft should never be negative:
`Debug.Assert(bytesLeft == 0);`

Reply:
The assert ensures bytesLeft is no negative